producer.go

```diff
@@ -215,26 +219,26 @@ func (p *Producer) loop() {
 	}
 }
 
-func (p *Producer) drainIfNeed() (*kinesis.PutRecordsRequestEntry, bool) {
+func (p *Producer) drainIfNeed() (ktypes.PutRecordsRequestEntry, bool) {
 	p.RLock()
 	needToDrain := p.aggregator.Size() > 0
 	p.RUnlock()
 	if needToDrain {
 		p.Lock()
-		record, err := p.aggregator.Drain()
+		record, has, err := p.aggregator.Drain()
```

Comment:
Let's not mix logic change with SDK upgrade. Also, if the previous code uses pointers, keep using pointers.